tests/infer/test_autoguide.py

```diff
@@ -140,8 +143,8 @@ def model():
 
 def auto_guide_list_x(model):
     guide = AutoGuideList(model)
-    guide.add(AutoDelta(poutine.block(model, expose=["x"])))
-    guide.add(AutoDiagonalNormal(poutine.block(model, hide=["x"])))
+    guide.x = AutoDelta(poutine.block(model, expose=["x"]))
```

Comment:
Should this be `.append(...)`?

Reply:
I wanted to check that both directly setting the attribute as well as `.append` works correctly. So both the code-paths are exercised by the tests.

Reply:
Actually, there are other guides the other code path; so I reverted this to `.append`.